### PR TITLE
Only install iOS simulator runtimes on builders that use them

### PR DESCRIPTION
**Pull request by jmagman**
iOS 13 and 15 runtime simulators are only needed for `Mac Unopt` at the moment.  Remove `runtime_versions` as default on Mac builders and only set the property where needed.  This should reduce the build time by ~4 minutes per builder.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: .ci.yaml
Comment (jmagman) [inline .ci.yaml:61]: @godofredoc do you think this is the right thing to do here to avoid the caches from colliding?  Hopefully we don't have that problem once you are done the symlinking fixes.

Comment (godofredoc) [reply .ci.yaml:61]: If I understand mac_toolchain correctly we don't need to differentiate between xcode and runtimes because they use different caches. In other words once the symlink changes land we will be able to keep a single cache per xcode version and single caches per runtime.

**Review by zanderso — APPROVED** [review]

**Comment by jmagman** [comment]
Holding off on this until @godofredoc is done his simulator symlink investigation.

**Review by godofredoc — APPROVED** [review]

**Pull request merged**
